install/ui/src/freeipa/plugins/certmap.js

```diff
@@ -0,0 +1,394 @@
+                        'cn',
+                        {
+                            $type: 'textarea',
+                            name: 'description'
+                        },
+                        'ipacertmapissuer',
+                        'ipacertmapmaprule',
+                        'ipacertmapmatchrule',
+                        {
+                            $type: 'multivalued',
+                            name: 'associateddomain'
+                        },
+                        'ipacertmappriority'
+                    ]
+                }
+            ]
+        }
+    ],
+    adder_dialog: {
+        $factory: certmap.rule_adder_dialog,
+        fields: [
+            'cn',
+            'ipacertmapissuer',
+            'ipacertmapmaprule',
+            'ipacertmapmatchrule',
+            {
+                $type: 'multivalued',
+                name: 'associateddomain'
+            },
+            'ipacertmappriority',
+            {
+                $type: 'textarea',
+                name: 'description'
+            }
+        ]
+    }
+};};
+
+
+var make_certmapconfig_spec = function() {
+return {
+    name: 'certmapconfig',
+    defines_key: false,
+    facets: [
+        {
+            $type: 'details',
+            facet_groups: [certmap.search_facet_group],
+            facet_group: 'search',
+            sections: [
+                {
+                    name: 'details',
+                    fields: [
+                        {
+                            $type: 'checkbox',
+                            name: 'ipacertmappromptusername'
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+};};
+
+
+/**
+ * Custom adder dialog for adding cert map rules which automatically turns on
+ * the rule - to be consistent with HBAC rules etc.
+ */
+certmap.rule_adder_dialog = function (spec) {
+    spec = spec || {};
+
+    var that = IPA.entity_adder_dialog(spec);
+
+    that.create_add_command = function(record) {
+        var command = that.entity_adder_dialog_create_add_command(record);
+
+        command.set_option('ipaenabledflag', true);
+
+        return command;
+    };
+
+    return that;
+};
+
+/**
+ * Multivalued widget which is used for working with user's certmap.
+ *
+ * @class
+ * @extends IPA.custom_command_multivalued_widget
+ */
+certmap.certmap_multivalued_widget = function (spec) {
+
+    spec = spec || {};
+
+    var that = IPA.custom_command_multivalued_widget(spec);
+
+    that.create_adder_dialog = function() {
+        var spec = that.adder_dialog_spec || {
+            name: 'custom-add-dialog',
+            title: '@i18n:objects.certmap.adder_title',
+            policies: [
+                {
+                    $factory: IPA.multiple_choice_section_policy,
+                    widget: 'type'
+                }
+            ]
+        };
+
+        that.adder_dialog = certmap.custom_adder_dialog(spec);
+        that.adder_dialog.create_button({
```

Comment:
Could we use confirmation mixin for the adder dialog? So that you would have to define the buttons? Or if it is not possible then define a dialog class for custom_command_multivalued_widget